Off-Top-App/off-top-flutter
# OT-34 | FE - Open a Websocket connection from Flutter.io to Java Spring

**What I Did:** Established communication between Spring-Boot server and Flutter through a WebSocket.

 **How To Test:** 
1. Go to off-top-flutter in your terminal and run `flutter packages get` to make sure you have all packages
2. if everything goes well run `flutter run` and wait until the emulator loads Off-Top
3. Now you will need to make sure your spring-boot server is running on OT-34 before proceeding.
- More info here: https://github.com/Off-Top-App/off-top-java/pull/8
4. You will see a button that says to test the `Go To WebSocket`, click on it, and you will redirected to a new page in mobile app where you can now on "Send Data to Weboskcet" to see that data has been received. It should like the photo in **Figure 1**
5. If you get the errors saying you're unable to connect to the websocket like the one in **Figure 2** you will need to change line 15 to `ws://10.0.2.2:8080/name`
### Figure 1 
<img width="438" alt="Screen Shot 2020-03-05 at 2 10 08 AM" src="https://user-images.githubusercontent.com/26909882/75971352-85b3c080-5e86-11ea-80d7-856c97565625.png">

### Figure 2
![port error](https://user-images.githubusercontent.com/26909882/75756092-d986a400-5ce4-11ea-99ae-881c5616f97a.JPG)
